localtime/data_source_timestamp_test.go

```diff
@@ -0,0 +1,68 @@
+package localtime
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+type TestAccDataSourceTimestampCase struct {
+	Name              string
+	Config            string
+	ExpectedTimestamp string
+}
+
+func TestAccDataSourceTimestamp_basic(t *testing.T) {
```

Comment:
Convert these into table tests, since they have the same structure. This will also help with running them in parallel, via `resource.ParallelTest`. Something like this:

```go
TestAccDataSourceTimestamp(t *testing.T) {
	cases := struct{
		Name              string
		Config            string
		ExpectedTimestamp string
	}{
		// 
	}

	for _, tc := range cases {
		tc := tc
		resource.ParallelTest(t, resource.TestCase{
			// 
		})
	}
}
```

The plugin SDK has moved away from mimicking Terraform and towards actually running the Terraform CLI. That's a significant source of the slowness. Ways around that are:

* Fewer independent test cases, with multiple resources per case. Terraform runs fewer times and is therefore faster.
* Parallel tests

Acceptance tests aren't meant to be particularly fast for a typical provider that needs an internet connection and real credentials. And parallelizing can be tricky and run through rate limits or other quotas. But if you're dealing with a local API or especially a wholly in memory API, it's worth changing the patterns a bit so test runs are faster.

Reply:
@bendrucker Having multiple tests in the loop sharing the same t will throw an error
```
panic: testing: t.Parallel called multiple times [recovered]
	panic: testing: t.Parallel called multiple times
```
I guess multiple test functions are still needed for parallel. See an example in [terraform-provider-aws](https://github.com/hashicorp/terraform-provider-aws/blob/d056371a65568504bfc6c5993a1fb993a0fc7759/aws/data_source_aws_ami_ids_test.go#L10-L26)